bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -182,6 +184,19 @@ public Integer getSample() {
                                               }
                                           });
         initializeExplicitLacFlushPolicy();
+
+        if (bk.getConf().getAddEntryQuorumTimeout() > 0) {
+            SafeRunnable monitor = new SafeRunnable() {
```

Comment:
I think if you use this [SafeRunnable](https://github.com/apache/bookkeeper/blob/master/bookkeeper-common/src/main/java/org/apache/bookkeeper/common/util/SafeRunnable.java), you can use lambda here.

Reply:
Unfortunately not. SafeRunnable is not a functional interface (I guess because it has more than one overridable methods).

Reply:
I had added something for that in MangedLedger code: 
https://github.com/apache/incubator-pulsar/blob/master/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/SafeRun.java

They way it's used is like: 

```java
executor.submit(safeRun(() -> {
 // ....
});
```

Reply:
The new `SafeRunnable` is a functional interface. The old `SafeRunnable` is going to be deprecated in future.

Reply:
Ah, I didn't know there were two SafeRunnables. Was in the middle of changing when @jiazhai merged :)

We can change these to use lambdas when we deprecate the old SafeRunnable.